### ISSUE TITLE
App crashes while starting CardEditActivity while passing the extras

### ISSUE THREAD

**Issue by Shajeel-Afzal** [sharish/CreditCardView#10]
I am starting the activity as:

```
Intent intent = new Intent(getActivity(), CardEditActivity.class);
                intent.putExtra(CreditCardUtils.EXTRA_CARD_HOLDER_NAME, "Shajeel Afzal");
                intent.putExtra(CreditCardUtils.EXTRA_CARD_NUMBER, "1234567890123456");
                intent.putExtra(CreditCardUtils.EXTRA_CARD_EXPIRY, "01/17");
                intent.putExtra(CreditCardUtils.EXTRA_CARD_CVV, "234");
                intent.putExtra(CreditCardUtils.EXTRA_CARD_SHOW_CARD_SIDE, CreditCardUtils.CARD_SIDE_BACK);

                startActivityForResult(intent, EDIT_CARD_CODE);
```

But the app crashes and shows the following log:

```
java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.EditText.selectAll()' on a null object reference
                                                                                at com.cooltechworks.creditcarddesign.pager.CardExpiryFragment.focus(CardExpiryFragment.java:122)
                                                                                at com.cooltechworks.creditcarddesign.pager.CardFragmentAdapter.focus(CardFragmentAdapter.java:14)
                                                                                at com.cooltechworks.creditcarddesign.CardEditActivity$3.onPageSelected(CardEditActivity.java:132)
                                                                                at android.support.v4.view.ViewPager.dispatchOnPageSelected(ViewPager.java:1931)
                                                                                at android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:662)
                                                                                at android.support.v4.view.ViewPager.setCurrentItemInternal(ViewPager.java:628)
                                                                                at android.support.v4.view.ViewPager.setCurrentItem(ViewPager.java:609)
                                                                                at com.cooltechworks.creditcarddesign.CardEditActivity.showNext(CardEditActivity.java:225)
                                                                                at com.cooltechworks.creditcarddesign.CardEditActivity$4.onCardEntryComplete(CardEditActivity.java:156)
                                                                                at com.cooltechworks.creditcarddesign.pager.CardFragmentAdapter.onActionComplete(CardFragmentAdapter.java:75)
                                                                                at com.cooltechworks.creditcarddesign.pager.CreditCardFragment.onComplete(CreditCardFragment.java:39)
                                                                                at com.cooltechworks.creditcarddesign.pager.CardNumberFragment.afterTextChanged(CardNumberFragment.java:71)
                                                                                at android.widget.TextView.sendAfterTextChanged(TextView.java:7738)
                                                                                at android.widget.TextView.setText(TextView.java:4105)
                                                                                at android.widget.TextView.setText(TextView.java:3954)
                                                                                at android.widget.EditText.setText(EditText.java:85)
                                                                                at android.widget.TextView.setText(TextView.java:3929)
                                                                                at com.cooltechworks.creditcarddesign.pager.CardNumberFragment.onCreateView(CardNumberFragment.java:43)
                                                                                at android.support.v4.app.Fragment.performCreateView(Fragment.java:2074)
                                                                                at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1104)
                                                                                at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1286)
                                                                                at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:758)
                                                                                at android.support.v4.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:1632)
                                                                                at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:637)
                                                                                at android.support.v4.app.FragmentStatePagerAdapter.finishUpdate(FragmentStatePagerAdapter.java:166)
                                                                                at android.support.v4.view.ViewPager.populate(ViewPager.java:1235)
                                                                                at android.support.v4.view.ViewPager.populate(ViewPager.java:1083)
                                                                                at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1609)
                                                                                at android.view.View.measure(View.java:17637)
                                                                                at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5536)
                                                                                at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1436)
                                                                                at android.widget.LinearLayout.measureVertical(LinearLayout.java:722)
                                                                                at android.widget.LinearLayout.onMeasure(LinearLayout.java:613)
                                                                                at android.view.View.measure(View.java:17637)
                                                                                at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5536)
                                                                                at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1436)
                                                                                at android.widget.LinearLayout.measureVertical(LinearLayout.java:722)
                                                                                at android.widget.LinearLayout.onMeasure(LinearLayout.java:613)
                                                                                at android.view.View.measure(View.java:17637)
                                                                                at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:728)
                                                                                at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:464)
                                                                                at android.view.View.measure(View.java:17637)
                                                                                at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5536)
                                                                                at android.widget.FrameLayout.onMeasure(FrameLayout.java:436)
                                                                                at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:135)
                                                                                at android.view.View.measure(View.java:17637)
                                                                                at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5536)
                                                                                at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1436)
                                                                                at android.widget.LinearLayout.measureVertical(LinearLayout.java:722)
                                                                                at android.widget.LinearLayout.onMeasure(LinearLayout.java:613)
                                                                            	at android.view.View.measure(Vie
```

**Comment by cooltechworks** [comment]
Shajeel, I tried to reproduce the issue and could not figure it out. Can you give me more specific information on which device/os you had this issue.

Thanks,
Harish

**Comment by Shajeel-Afzal** [comment]
Hi Harish!

I am testing it on Android 6.

**Comment by Shajeel-Afzal** [comment]
I have found that when I do not pass the credit card number it works without any problem. But passing `intent.putExtra(CreditCardUtils.EXTRA_CARD_NUMBER, "5365678900000000");` crashes the app. 

Maybe i am passing the card number in the wrong format?

**Comment by cooltechworks** [comment]
Your format is good. Let me check it.

**Comment by glarence** [comment]
I recently just tried using this custom UX and came across the same issue. Is there a fix?

**Comment by sharish** [comment]
This will be fixed in the upcoming version! 

Thanks,
Harish

**Comment by glarence** [comment]
Thank you! Just an FYI, I have my project due by 10/31/16 Pacific Time, but take as much time as you need! 

Get Outlook for iOS




On Thu, Oct 27, 2016 at 8:46 PM -0700, "Harish Sridharan" <notifications@github.com> wrote:












This will be fixed in the upcoming version! 



Thanks,

Harish

**Comment by sharish** [comment]
Fixed in v1.0.2.

Thanks,
Harish
